### PR TITLE
feat: pure javascript impl murmurhash v2. Thanks @zhangzifa

### PR DESCRIPTION
**Pull request by fengmk2**
```bash
$ node test/benchmark.js

murmurhash2js 3389647783
murmurhash 3389647783
murmurhash2js 4040304760
murmurhash 4040304760
murmurhash2js utf8 3300337389
murmurhash utf8 3300337389
murmurhash('hello 中国'): 1248731102
murmurhash2js('hello 中国'): 1248731102

node version: v1.7.1, date: Mon Apr 20 2015 17:44:59 GMT+0800 (CST)
Starting...
4 tests completed.

murmurhash(ascii)    x 8,677,245 ops/sec ±0.94% (98 runs sampled)
murmurhash2js(ascii) x 7,800,580 ops/sec ±3.29% (89 runs sampled)
murmurhash(utf8)     x 7,154,121 ops/sec ±2.38% (90 runs sampled)
murmurhash2js(utf8)  x 4,565,477 ops/sec ±1.24% (90 runs sampled)
```

### CODE REVIEW

**Comment by fengmk2** [comment]
Thanks @zhangzifa and @JacksonTian

File: test/benchmark.js [hunk 1]
```diff
@@ -44,11 +82,11 @@ function murmurhash2js(key, seed) {
 
   switch (l) {
     case 3:
-      h ^= (key[i + 2] & 0xff) << 16;
+      h ^= NEGTIVE_MAP[key[i + 2]] << 16;
```
Comment (fengmk2) [inline test/benchmark.js]: @coolme200

File: test/benchmark.js [hunk 2]
```diff
@@ -17,6 +17,44 @@ var murmurhash = require('../');
 
 var MURMURHASH_M = 0x5bd1e995;
 
+// @zhangzifa: 大于127的数, 在c里面的负数, 表示成0xffffffxx才成
```
Comment (JacksonTian) [inline test/benchmark.js]: benchmark里不需要再定义了吧

Comment (fengmk2) [inline test/benchmark.js]: 方便看 diff，就留着吧

**Comment by JacksonTian** [comment]
还是多加点测试数据。

**Comment by fengmk2** [comment]
呃，不是已经找到算法差异点了么，找到了就没必要加了。

File: test/murmurhash.test.js [hunk 3]
```diff
@@ -42,9 +45,82 @@ describe('murmurhash.test.js', function () {
...
+    murmurhash(buf).should.equal(2673451384);
+  });
+
+  it('should emtpy string got 3397915750', function () {
+    murmurhash('').should.equal(3397915750);
+    murmurhash(new Buffer('')).should.equal(3397915750);
+    murmurhash(new Buffer([])).should.equal(3397915750);
+  });
+
+  it('should work with boundary cases: buffer mod length equal to 0,1,2,3', function () {
```
Comment (fengmk2) [inline test/murmurhash.test.js]: @zhangzifa done

**Comment by fengmk2** [comment]
@yjhjstz 现在这样使用有问题？

**Comment by JacksonTian** [comment]
这块数据没多大。不用担心长期驻留的问题。

File: lib/murmur.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  0xfffffff8, 0xfffffff9, 0xfffffffa, 0xfffffffb, 0xfffffffc, 0xfffffffd,
+  0xfffffffe, 0xffffffff
+];
+
+function murmurhash2js(key, seed) {
+  if (typeof seed !== 'number') {
+    seed = 97;
+  }
+  var l = key.length;
+  var h = seed ^ l;
```
Comment (JacksonTian) [inline lib/murmur.js]: 这里的初始值会不会有边界问题？ @zhangzifa

**Comment by yjhjstz** [comment]
可以用Buffer替代吧

**Comment by JacksonTian** [comment]
不能

发自我的 iPhone

**Comment by fengmk2** [comment]
@yjhjstz 为什么还会被 GC 呢？

**Comment by JacksonTian** [comment]
常驻对象会在老生代中，会被扫描到，但不会被回收。他指的是会被扫描，比较浪费扫描时间。

**Comment by fengmk2** [comment]
那像 module.exports = xxx 的这种，都浪费扫描时间了。。。

**Comment by JacksonTian** [comment]
嗯。
